packages/sol6/src/yang/etsi-nfv-ns.yang

```diff
@@ -148,31 +154,61 @@ submodule etsi-nfv-ns {
           }
           description
             "References the descriptor of the NS VL instance to
-             which the SAP instantiated from this SAPD connects to.";
+            which the SAP instantiated from this SAPD connects to.";
           reference
             "GS NFV IFA014: Section 6.2.3.2 Sapd information element";
         }
 
-	choice associated-cpd-id {
-	  leaf vnf-ext-cpd {
-	    type leafref {
-	      path "deref(../virtual-link-desc)/../../../vnfd/ext-cpd/id";
-	    }
-	  }
-	  leaf pnf-ext-cpd {
-	    type leafref {
-	      path "deref(../virtual-link-desc)/../../../pnfd/ext-cpd/id";
-	    }
-	  }
-	  leaf sapd {
-	    type leafref {
-	      path "../../id";
-	    }
-	  }
-	}
-
-	uses common:cpd;
+        choice associated-cpd-id {
+            container vnf {
+              leaf vnfd-id {
+                mandatory true;
+                type leafref {
+                   path "/nfv:nfv/nfv:vnfd/nfv:id";
+                }
+                must "boolean(../../../vnfd-id[.=current()])";
```

Comment:
I do not understand this must statement. Isn't current() supposed to return the context of the current node, which is vnfd-id?

Reply:
It says that the current leaf called vnfd-id must exist in the leaflist "above" also called vnfd-id.

So current() points to the leaf we see here and "." is the value of the target leaflist. 

You have to love XPATH... :)